util/network-devp2p/src/discovery.rs

```diff
@@ -484,6 +490,11 @@ impl Discovery {
 			PACKET_PONG => self.on_pong(&rlp, node_id, from),
 			PACKET_FIND_NODE => self.on_find_node(&rlp, node_id, from),
 			PACKET_NEIGHBOURS => self.on_neighbours(&rlp, node_id, from),
+			PACKET_ENR_REQUEST => self.on_enr_request(&rlp, node_id, from, hash_signed.as_bytes()),
+			PACKET_ENR_RESPONSE => {
+				debug!(target: "discovery", "ENR response handling is not implemented");
```

Comment:
Remind me, what is the challenge here? What is it we're supposed to use the ENR responses for again?

Reply:
I don't know? I think they will be needed for Discovery v5, but right now ENR is a piece of state of value to other consumers.

Reply:
> The recipient of the packet should verify that the node record is signed by node who sent ENRResponse.

I can't find that that is performed, can you explain?

Reply:
But we don't request ENRs here and thus don't handle ENR responses from peers, no?

Reply:
It is probably fine to keep out it of this PR.